### PR TITLE
add new operator and fix eol issue

### LINKED ISSUE #144: [lexer] New operator
**Subject**
Add new operator

**Detail**
- [ ] +=
- [ ] -=
- [ ] *=
- [ ] /=
- [ ] %=
- [ ] &&
- [ ] ||
- [ ] ++
- [ ] --

### PR DESCRIPTION
**Pull request by joomanzi**
resolved: #144 #138 #186

1. add new operator and refactoring defaultStateFn
+=, -=, *=, /=, %=, &&, ||, ++, --
2. Add insertSemi val and fix eol issue

add insertSemi value in stateFn and solve EOL issue in lexer.
now lexer don't return EOL token.
semicolon added when final token at end of line(\n) is

- Number, String, Ident
- ++, --
- } )
it consume every comment block or line
- [ ] Test case

### CODE REVIEW

File: parse/lex.go [hunk 1]
```diff
@@ -254,66 +255,100 @@ func (s *state) acceptRun(valid string) {
 //	Rbrace // }
 //
 
+func (s *state) isNextToken(next rune) bool {
+	if s.peek() == next {
+		s.next()
+		return true
+	} else {
+		return false
+	}
```
Comment (boohyunsik) [inline parse/lex.go:262]: How about think this?
```go
if s.peek() == next {
    s.next()
    return true
}
return false

File: parse/token.go [hunk 2]
```diff
@@ -111,6 +123,17 @@ var TokenTypeMap = map[TokenType]string{
 	Slash:    "SLASH",
 	Mod:      "MOD",
 
+	Plus_assign:     "PLUS_ASSIGN",
```
Comment (boohyunsik) [inline parse/token.go]: In ast.go, we are using camel type (like PlusAssign).
I think we should use same type.
How about this?

Comment (joomanzi) [reply parse/token.go]: appreciate u Mr.Boo

**Review by junbeomlee — APPROVED** [review]
LGTM!

**Pull request merged**